src/log-pcap-stream.c

```diff
@@ -0,0 +1,311 @@
+ *  \param data Populated with the thread data structure.
+ *  \return TM_ECODE_OK On success.
+ *  \return TM_ECODE_FAILED On serious error.
+ */
+static TmEcode StreamPcapLogThreadInit(ThreadVars *tv, const void *initdata,
+        void **data)
+{
+    // Create and initialize the thread data.
+    if (initdata == NULL) {
+        SCLogDebug("Error getting context for StreamLogPcap. \"initdata\" "
+                   "argument NULL");
+        return TM_ECODE_FAILED;
+    }
+    StreamOutputCtx *stream_output_ctx = getStreamOutputCtx((OutputCtx*)
+            initdata);
+    StreamPcapLogThreadData *td = SCMalloc(sizeof(*td));
+    if (unlikely(td == NULL)) {
+        return TM_ECODE_FAILED;
+    }
+    td->stream_output_ctx = stream_output_ctx;
+    td->count = 0;
+    *data = (void *)td;
+    return TM_ECODE_OK;
+}
+
+/**
+ * \brief Thread deinit function.
+ * \param thread_data StreamPcapLog thread data.
+ * \return TM_ECODE_OK On success.
+ */
+static TmEcode StreamPcapLogThreadDeInit(ThreadVars *tv, void *thread_data)
+{
+    StreamPcapLogThreadData *td = thread_data;
+    SCFree(td);
+    return TM_ECODE_OK;
+}
+
+/**
+ *  \brief Fill in stream-pcap logging struct from the provided ConfNode.
+ *  \param conf The configuration node for this output.
+ *  \return output_ctx: Output context (contains a StreamOutputCtx data member)
+ */
+static OutputInitResult StreamPcapLogInitCtx(ConfNode *conf)
+{
+    SCLogNotice("StreamPcapLogInitCtx enter");
+    OutputInitResult result =  {NULL, false};
+    /* Create the output context from the configuration node. */
+    StreamOutputCtx *stream_output_ctx =
+            SCMalloc(sizeof(*stream_output_ctx));
+    if (unlikely(stream_output_ctx == NULL)) {
+        SCReturnCT(result, "OutputInitResult");
+    }
+    OutputCtx *output_ctx = SCMalloc(sizeof(*output_ctx));
+    if (unlikely(output_ctx == NULL)) {
+        SCFree(stream_output_ctx);
+        SCReturnCT(result, "OutputInitResult");
+    }
+    /* Initialize filename */
+    stream_output_ctx->filename[0] = '\0';
+
+    /* Load output options. */
+    const char *outputDirectory = ConfNodeLookupChildValue(conf,
+            "output-directory");
+
+    InitializePcapLogFilenameSupport(outputDirectory);
+
+    if (ConfNodeChildValueIsTrue(conf, "session-dump")) {
+        stream_output_ctx->session_dump_enabled = true;
+    } else {
+        stream_output_ctx->session_dump_enabled = false;
+    }
+    output_ctx->data = stream_output_ctx;
+    output_ctx->DeInit = StreamPcapLogFileDeInitCtx;
+    result.ctx = output_ctx;
+    result.ok = true;
+    SCReturnCT(result, "OutputInitResult");
+}
+
+/**
+ *  \brief Helper function to extract the StreamOutputCtx from an
+ *   OutputCtx->data
+ *  \param output_ctx Structure that output modules use to maintain private data
+ *  \return *StreamOutputCtx on success.
+ *  \return NULL on failure or if output_ctx->data is NULL.
+ */
+static StreamOutputCtx *getStreamOutputCtx(const OutputCtx *output_ctx)
```

Comment:
style: GetStream... (start with capital)

Reply:
Will fix this up.